src/logic/wallets/ethTransactions.ts

```diff
@@ -38,18 +39,11 @@ export const calculateGasPrice = async (): Promise<string> => {
   return Promise.reject(err)
 }
 
-export const calculateGasOf = async (txConfig: {
-  to: string
-  from: string
-  data: string
-  gasPrice?: number
-  gas?: number
-}): Promise<number> => {
-  const web3 = getWeb3()
+export const calculateGasOf = async (txConfig: EthAdapterTransaction): Promise<number> => {
   try {
-    const gas = await web3.eth.estimateGas(txConfig)
+    const ethAdapter = getSDKWeb3Adapter(txConfig.from)
 
-    return gas
+    return await ethAdapter.estimateGas(txConfig)
   } catch (err) {
     return Promise.reject(err)
```

Comment:
Lol. A very useful try-catch. Maybe we could throw a custom error code here?

Reply:
I've added one in my latest commit.